src/main/java/com/lirc572/ip/Parser.java

```diff
@@ -0,0 +1,218 @@
+package com.lirc572.ip;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Contains static methods for parsing commands.
+ */
+public class Parser {
+
+    /**
+     * All exit commands are listed here.
+     */
+    private static final Set<String> EXIT_COMMANDS = Set.of(
+            "bye", "exit", "quit"
+    );
+
+    /**
+     * Tokenizes a command.
+     *
+     * @param command A line of command.
+     * @return The tokenized command as a String array.
+     */
+    public static String[] tokenizeCommand(String command) {
+        ArrayList<String> tokens = new ArrayList<>();
+        boolean isNewToken = true;
+        boolean unmatchedQuote = false;
+        char quotationMark = ' ';
+        for (char ch : command.toCharArray()) {
+            if (!unmatchedQuote) {
+                // if not a whitespace character
+                if (ch != ' ' && ch != '\t') {
+                    // if right after a whitespace character
+                    if (isNewToken) {
+                        // start of a quoted token
+                        if (ch == '"' || ch == '\'') {
+                            quotationMark = ch;
+                            unmatchedQuote = true;
+                            tokens.add("");
+                        } else {
+                            tokens.add(String.valueOf(ch));
+                            isNewToken = false;
+                        }
+                    } else {
+                        tokens.set(tokens.size() - 1, tokens.get(tokens.size() - 1) + ch);
+                    }
+                } else {
+                    isNewToken = true;
+                }
+            } else {
+                if (ch != quotationMark) {
+                    tokens.set(tokens.size() - 1, tokens.get(tokens.size() - 1) + ch);
+                } else {
+                    unmatchedQuote = false;
+                    isNewToken = true;
+                }
+            }
+        }
+        String[] result = new String[tokens.size()];
+        return tokens.toArray(result);
+    }
+
+    /**
+     * Processes a command.
+     *
+     * @param command The command to process.
+     * @return Whether the program should continue (<code>true</code> if not an exit command).
+     */
+    public static boolean processCommand(String command, TaskList tasks) throws Exception {
```

Comment:
This method could possibly be broken up as well and could better adhere to Java Coding Standards